src/dfm/setup.rs

```diff
@@ -0,0 +1,25 @@
+use super::model::Config;
+use anyhow::{Result, anyhow};
+use std::fs;
+use std::path::PathBuf;
+
+pub fn setup(path: PathBuf) -> Result<()> {
+    let config = generate_config(path)
+
+    println!("{:?}", config?);
```

Comment:
One thing to call out here -- `"{config:?}"` will call the `fmt::Debug::fmt` trait function. (You derived this trait on your config type). If you implement [`fmt::Display`](https://doc.rust-lang.org/std/fmt/trait.Display.html) you can use `"{config}"` or `println!("{}", config)` (note: there's no `:?` in the braces). This allows you to control how the object prints and maybe even make it prettier for display purposes. Just something to consider.